### PR TITLE
Nginx proxy for workroom

### PR DESCRIPTION
**Pull request by jfryman**
This PR introduces `nginx` as a proxy to serve out `st2auth` and provide PAM integration, `st2api`, `st2installer`, and `mistral`. Each of these applications are WSGI applications that need a common serve point.

This introduces several new things:
- [x] SSL Cert Generation. A user can either provide a SSL certificate at run-time, or have one automatically generated. Either way, all transmissions to and from will be encrypted.
- [x] PAM based authentication. Users with access to the machine will also be granted access to StackStorm.
- [x] Nginx reverse proxy. Nginx has been configured with SSL lockdown to ensure highest grade encryption out of the box. 
- [x] NTP - Clocks drift. Now that SSL is in the picture, that's not going to be allowed. Includes ability to keep up-to-date.
- [x] `uwsgi` server. All of the aforementioned applications are managed by `uwsgi` in Emperor mode. Watches for configuration from each of the apps on where and how to bootstrap, and then does the needful.
- [x] st2installer. New and improved, has a nice little thing that is used to configure new installations of StackStorm. Neato!

### CODE REVIEW

File: modules/profile/manifests/st2server.pp [hunk 1]
```diff
@@ -1,4 +1,286 @@
...
+      source   => 'https://github.com/stackstorm/st2installer',
+      require  => Class['::st2::profile::server'],
+    }
+
+    uwsgi::app { 'st2installer':
+      ensure              => present,
+      uid                 => $_nginx_daemon_user,
+      gid                 => $_nginx_daemon_user,
+      application_options => {
+        'http-socket'  => "127.0.0.1:${_st2installer_port}",
```
Comment (DoriftoShoes) [inline modules/profile/manifests/st2server.pp:262]: Did we go with an http socket over a standard unix socket for any particular reason?

`'socket' => '/tmp/uwsgi_st2installer.sock'`

Comment (jfryman) [inline modules/profile/manifests/st2server.pp:262]: Where is that? Everything should be a TCP port now on `127.0.0.1`.

The only reason a socket was in there in the first place was when I was working out how to manage ports all over the place. Once I got config ability to manage `st2api` and `st2auth`, I reverted back to ports.

Matches the future Docker state. The hope was that most of this work is re-use once we flip to containers, so I tried to treat the connections the same way.

Comment (DoriftoShoes) [inline modules/profile/manifests/st2server.pp:262]: It's just my personal preference but I would rather serve uwsgi on a local file socket and proxy to that.  I guess serving http to localhost might help in some debugging situations but I find it a bit confusing when looking at netstat and stuff personally.

Comment (DoriftoShoes) [inline modules/profile/manifests/st2server.pp:262]: This is how I was using uwsgi with mistral api:
`uwsgi -s /opt/stackstorm/mistral_api.sock --wsgi-file <MISTRAL>/mistral/api/wsgi.py -H /opt/openstack/mistral/.venv/`

nginx config:
https://gist.githubusercontent.com/DoriftoShoes/d3ae68f0b3a9ce77a163/raw/feb7d57dcf88dc47210b02619e2ae5129447621c/mistral.conf

**Comment by DoriftoShoes** [comment]
LGTM!

**Pull request merged**